rosidl_generator_py/resource/_msg.py.template

```diff
@@ -98,7 +102,7 @@ class @(spec.base_type.type)(metaclass=Metaclass):
         self.@(field.name) = kwargs.get(
             '@(field.name)', @(spec.base_type.type).@(field.name.upper())__DEFAULT)
 @[    else]@
-@[      if not field.type.is_primitive_type()]@
+@[      if not field.type.is_primitive_type() and (not field.type.is_array or field.type.array_size)]@
```

Comment:
You're right, the import I was looking for (for all non primitive types) is in Line 131/135.